onnxruntime/python/tools/transformers/fusion_bart_attention.py

```diff
@@ -181,12 +327,18 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
         qk_nodes_2 = self.model.match_parent_path(
             matmul_qkv, ["Softmax", "Reshape", "Add", "Reshape", "MatMul"], [0, 0, 0, 0, 0]
         )
+        qk_nodes_2_openai = self.model.match_parent_path(matmul_qkv, ["Softmax", "Add", "MatMul"], [0, 0, 0])
+        add_qk = None
         if qk_nodes_1 is not None:
             _, matmul_qk = qk_nodes_1
             qk_nodes = qk_nodes_1
         elif qk_nodes_2 is not None:
             _, _, add_qk, _, matmul_qk = qk_nodes_2
             qk_nodes = qk_nodes_2
+        elif qk_nodes_2_openai is not None:
+            _, add_qk, matmul_qk = qk_nodes_2_openai
+            qk_nodes = qk_nodes_2_openai
+            qk_nodes_2 = qk_nodes_2_openai
```

Comment:
Is it possible to remove this line and instead modify the code that identifies the 5 types of attention in the model?

https://github.com/microsoft/onnxruntime/blob/d611b2e81933294e1175ce5702a79e458781516d/onnxruntime/python/tools/transformers/fusion_bart_attention.py#L527-L539

Ex: `decoder_attention = one_root_input and (qk_nodes == qk_nodes_2 or qk_nodes == qk_nodes_2_openai)`